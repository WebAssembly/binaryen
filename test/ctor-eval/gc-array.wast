(module
  (type $array (array (mut i32)))

  (import "import" "import" (func $import (param anyref)))

  ;; This global will remain as it is.
  (global $global1 (ref $array)
    (array.new_fixed $array
      (i32.const 10)
      (i32.const 20)
      (i32.const 30)
      (i32.const 40)
    )
  )

  (global $global2 (ref $array)
    (array.new_fixed $array
      (i32.const 42)
      ;; This location will be written with a new value, 1337
      (i32.const 0)
    )
  )

  (func "test1"
    (array.set $array
      (global.get $global2)
      (i32.const 1)
      (i32.const 1337)
    )
  )

  (func "keepalive" (result i32)
    (i32.add
      (array.get $array
        (global.get $global1)
        (i32.const 0)
      )
      (array.get $array
        (global.get $global2)
        (i32.const 0)
      )
    )
  )
)

