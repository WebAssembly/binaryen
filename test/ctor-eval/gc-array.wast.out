(module
 (type $array (array (mut i32)))
 (type $none_=>_i32 (func (result i32)))
 (global $global1 (ref $array) (array.new_fixed $array
  (i32.const 10)
  (i32.const 20)
  (i32.const 30)
  (i32.const 40)
 ))
 (global $global2 (ref $array) (array.new_fixed $array
  (i32.const 42)
  (i32.const 1337)
 ))
 (export "keepalive" (func $1))
 (func $1 (type $none_=>_i32) (result i32)
  (i32.add
   (array.get $array
    (global.get $global1)
    (i32.const 0)
   )
   (array.get $array
    (global.get $global2)
    (i32.const 0)
   )
  )
 )
)
